### DISCUSSION TITLE
Personal patches for Labwc

### DISCUSSION THREAD

**Discussion by droc12345** [labwc/labwc#2131] [Show and tell]
I have started to carry some patches for labwc @ https://github.com/droc12345/images

As of right now, I have 3
1. Decorate/Undecorate that mirrors openbox, and BorderOnly that creates just a border.
these are not toggle values, they simply set what they are called with, a simpler variation of SetDecorate
that doesn't use keepBorder to make decisions, use with next patch for openbox "feel".
2. Convert ToggleDecorate to 2 state like openbox
3. Patch to add client states (min, max, full, active) to client-list-combined-menu

Left and right are pipemenus, left is regular pipemenu, right is created by a modified labwc-menu-generator
the middle one is client-list-combined-menu as submenu of root menu, but can be bound to a key for easy use
I like it bound to alt-space to easily bring it up.
![menus](/../../../droc12345/images/blob/main/data/menu-aug-2024.jpg?raw=true)

Since I use these on my personal system, I do keep them up to date with the latest git.
Enjoy

**Comment by droc12345** [comment]
I'm going to close this for now, client list combined menu has been merged, and I expect the send to menu to be added 
in the not too distant future.

**Comment by droc12345** [comment]
Removed the built in menus patch as it has been merged.

Added a patch to provide states for client list combined menu.

**Comment by davidphilipbarr** [comment]
Was wondering  if it would useful or more confusing to have the 'go there...' not on the current desktop? like:

`if (workspace != server->workspace_current) { ... }` ?

also wondering if a separator would be helpful above the go there?

**Comment by droc12345** [comment]
I could have made it skip the "go there" for current, but decided the logic wasn't worth adding in.
Perhaps a better name than "Go there" might be useful, but anyone coming from openbox should understand that as that's 
what it uses. Could use a separator, but that starts eating up screen space, and I haven't implemented scrolling yet.

I know openbox doesn't show "go there" unless there are no apps running in that workspace,
but I wanted the ability to swap to a workspace without disturbing the states of the windows. 

Edit to add: with the states patch, I moved over the title by a few spaces, so they stand out from the Go there at the bottom.
